File: ARC160/a.py
n , k = map(int, input().split())
lst = list(map(int, input().split()))
keys = []
for i_l in range(1, n+1):
    for i_r in range(1, n+1):
        if i_l <= i_r:
            keys.append((i_l, i_r))


def make_rev_numbers(l, r):
    left_part = lst[:l]
    before_rev = lst[l:r+1]
    rev = before_rev[::-1]
    right_part = lst[r+1:]
    ret = left_part + rev + right_part
    return ret


nums = []
for pair in keys:
    rev_lst = make_rev_numbers(pair[0]-1, pair[1]-1)
    cat_num = int(''.join(map(str, rev_lst)))
    nums.append([cat_num, rev_lst, ] )

sorted_nums = sorted(nums,  key=lambda x: x[0])
print(*sorted_nums[k-1][1])

File: ARC160/a.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int compare(const void * a, const void * b) {
    return *(int*)a - *(int*)b;
}

int main() {
    int n, k;
    scanf("%d %d", &n, &k);

    int lst[n];
    for(int i = 0; i < n; i++) {
        scanf("%d", &lst[i]);
    }

    int keys[n*n][2];
    int index = 0;
    for(int i_l = 1; i_l <= n; i_l++) {
        for(int i_r = 1; i_r <= n; i_r++) {
            if(i_l <= i_r) {
                keys[index][0] = i_l;
                keys[index][1] = i_r;
                index++;
            }
        }
    }

    int nums[n*n][n+1];
    for(int i = 0; i < index; i++) {
        int l = keys[i][0] - 1;
        int r = keys[i][1] - 1;
        int rev_lst[n];
        memcpy(rev_lst, lst, sizeof(int)*n);

        for(int j = 0; j < (r-l+1)/2; j++) {
            int tmp = rev_lst[l+j];
            rev_lst[l+j] = rev_lst[r-j];
            rev_lst[r-j] = tmp;
        }

        int cat_num = 0;
        for(int j = 0; j < n; j++) {
            cat_num = cat_num * 10 + rev_lst[j];
        }

        nums[i][0] = cat_num;
        for(int j = 0; j < n; j++) {
            nums[i][j+1] = rev_lst[j];
        }
    }

    qsort(nums, index, sizeof(int)*(n+1), compare);

    for(int i = 0; i < n; i++) {
        printf("%d ", nums[k-1][i+1]);
    }

    return 0;
}
